model/ManyManyList.php

```diff
@@ -69,8 +69,11 @@ public function __construct($dataClass, $joinTable, $localKey, $foreignKey, $ext
 	 */
 	protected function linkJoinTable() {
 		// Join to the many-many join table
-		$baseClass = ClassInfo::baseDataClass($this->dataClass);
-		$this->dataQuery->innerJoin($this->joinTable, "\"{$this->joinTable}\".\"{$this->localKey}\" = \"{$baseClass}\".\"ID\"");
+		$dataClassIDColumn = DataObject::getSchema()->quotedColumn($this->dataClass(), 'ID');
+		$this->dataQuery->innerJoin(
+			$this->joinTable,
+			"\"{$this->joinTable}\".\"{$this->localKey}\" = {$dataClassIDColumn}"
```

Comment:
Doesn't work on tables (e.g. many_many maps) that don't have DataObject backing.